### DISCUSSION TITLE
Error - Failed to scrape resource for metric

### DISCUSSION THREAD

**Discussion by ayouba88** [tomkerkhove/promitor#1822] [Q&A]
Hello,

I have confguired scraper with an resource group and a resource and confguired all the needed metrics
but for most of the metrics i am getting the below message in the scraper log
all the metrics have the value 0 for now. would that be the root cause for such an error ? 

```
[20:57:01 FTL] Failed to scrape resource for metric 'IfUnderDDoSAttack'
System.ArgumentException: [details] cannot be Null, empty or white-space. (Parameter 'details')
   at GuardNet.Guard.For[TException](Func`1 predicate, TException exception)
   at Promitor.Integrations.AzureMonitor.Exceptions.MetricInformationNotFoundException..ctor(String name, String details, String dimension) in /src/Promitor.Integrations.AzureMonitor/MetricInformationNotFoundException.cs:line 36
   at Promitor.Integrations.AzureMonitor.AzureMonitorClient.QueryMetricAsync(String metricName, String metricDimension, AggregationType aggregationType, TimeSpan aggregationInterval, String resourceId, String metricFilter) in /src/Promitor.Integrations.AzureMonitor/AzureMonitorClient.cs:line 85
   at Promitor.Core.Scraping.AzureMonitorScraper`1.ScrapeResourceAsync(String subscriptionId, ScrapeDefinition`1 scrapeDefinition, TResourceDefinition resourceDefinition, AggregationType aggregationType, TimeSpan aggregationInterval) in /src/Promitor.Core.Scraping/AzureMonitorScraper.cs:line 69
  ​at Promitor.Core.Scraping.Scraper`1.ScrapeAsync(ScrapeDefinition`1 scrapeDefinition) in /src/Promitor.Core.Scraping/Scraper.cs:line 89
```

Config:
```yaml
azureMetadata:
  subscriptionId: ******************************
  tenantId: *****************************
  resourceGroupName: ********my-resource-group**********
metricDefaults:
  aggregation:
    interval: 00:05:00
  scraping:
    schedule: 0 * * ? * *
metrics:
- name: IfUnderDDoSAttack
  description: "Under DDoS attack or not"
  resourceType: Generic
  azureMetricConfiguration:
    metricName: IfUnderDDoSAttack
    aggregation:
      type: Maximum
  resources:
  - resourceUri: Microsoft.Network/virtualNetworks/********my-resource**********
```

best regards,
Ayoub

**Comment by tomkerkhove** [comment]
The metric that you have configured does not exist (`IfUnderDDoSAttack`), the error can be improved though.

**Reply by LeonidShamis** [reply]
Hi. I can see the **IfUnderDDoSAttack** metric defined for the [https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-supported#**microsoftnetworkvirtualnetworks**](https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-supported#microsoftnetworkvirtualnetworks)

**Reply by ayouba88** [reply]
Hi Tom,

even if it is not defined, but i have the same error for many other metrics - see below example

- name: BytesDroppedDDoS
  description: "Inbound bytes dropped DDoS"
  resourceType: Generic
  azureMetricConfiguration:
    metricName: BytesDroppedDDoS
    aggregation:
      type: Maximum
  resources:
  - resourceUri: Microsoft.Network/virtualNetworks/*********my-resource-*************

same for 
'BytesDroppedDDoS'
'BytesForwardedDDoS'
'BytesInDDoS'
'DDoSTriggerSYNPackets'
'DDoSTriggerTCPPackets'
'DDoSTriggerUDPPackets'
'IfUnderDDoSAttack'
'PacketsDroppedDDoS'
'PacketsForwardedDDoS'
'PacketsInDDoS'
'PeeringAvailability'
'TCPBytesDroppedDDoS'
'TCPBytesForwardedDDoS'
'TCPBytesInDDoS'
'TCPPacketsDroppedDDoS'
'TCPPacketsForwardedDDoS'
'TCPPacketsInDDoS'
'UDPBytesDroppedDDoS'
'UDPBytesForwardedDDoS'
'UDPBytesInDDoS'
'UDPPacketsDroppedDDoS'
'UDPPacketsForwardedDDoS'
'UDPPacketsInDDoS'


could it be version related ? 
reminder, 
promitor-agent-scraper                 2.1.0
promitor-agent-resource-discovery      0.2.0

**Reply by tomkerkhove** [reply]
Odd, can you try using our latest agents please?
